packages/client/src/components/ModuleTree/Module/Icons/Icon.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { PropsWithRef } from 'react';
+import styled from 'styled-components';
+
+import { IconFire } from '@codement/ui/components/Icon/icons/solid/cis-fire';
+import { IconCheck } from '@codement/ui/components/Icon/icons/solid/cis-check';
+
+import { Circle } from './circle';
+import { Hexagon } from './hexagon';
+import { Octagon } from './octagon';
+
+/* TODO: icon.tsx is modified based on ui/icon.
+* If the way to move to style-component method is approved, we can move these to ui/icon.
+*/
+export const icons = {
+  circle: Circle,
+  hexagon: Hexagon,
+  octagon: Octagon,
+  fire: IconFire,
+  check: IconCheck
+};
+
+export type IconType = keyof typeof icons;
+
+export interface IconProps extends PropsWithRef<any> {
+  icon: IconType;
+  className: string;
+  size: 'large' | 'small' | 'medium' | number;
+  position: string;
+  left:string;
+  top:string;
+  height: string;
+  fill: string;
+  stroke: string;
+  transform: string;
+}
+
+export const sizeMap = {
+  small: '1',
+  medium: '2.5',
+  large: '5'
+};
+
+// TODO: there is lint error for all the props: xxx is defined but never used.
+export const Icon: React.FunctionComponent<IconProps> = ({
```

Comment:
You are repeating yourself here creating the same icon we already have in UI, can you use the UI one and adapt it if needed?